docs/diagrams/wallet-lifecycle/wallet-lifecycle-3-responding-to-moving-funds-state.tikz

```diff
@@ -0,0 +1,153 @@
+\node[note2, above=of rmft_note1.north, anchor=south, yshift=-8mm] (rmft_note2)   {Sender incentivized by:\\ \textbf{Reimbursement, if maintainer}};
+
+% Arrows
+\draw[->] (rmft_start)   -- (rmft_check1);
+\draw[->] (rmft_check1)  -| (rmft_revert2) \no;
+\draw[->] (rmft_check1)  -| (rmft_check2)  \yes;
+\draw[->] (rmft_check2)  -| (rmft_check3)  \no;
+\draw[->] (rmft_check2)  -| (rmft_revert1) \yes;
+\draw[->] (rmft_check3)  -| (rmft_revert2) \no;
+\draw[->] (rmft_check3)  -| (rmft_check4)  \yes;
+\draw[->] (rmft_check4)  -| (rmft_revert2) \no;
+\draw[->] (rmft_check4)  -| (rmft_action1) \yes;
+\draw[->] (rmft_action1) -- (rmft_action2);
+
+% MOVING FUNDS COMMITMENT
+% Nodes
+\node[start, right=of rmft_start, xshift=12cm]                    (mfc_start)   {Somebody sends \textbf{moving funds commitment}};
+\node[check, below=of mfc_start]                                  (mfc_check1)  {Is the\\ source wallet in\\ MOVING\_FUNDS\\ state?};
+\node[check, below right=of mfc_check1]                           (mfc_check2)  {Is sender a\\ member of the\\ source wallet?};
+\node[check, below right=of mfc_check2]                           (mfc_check3)  {Hasn't the\\ target\_wallet\_commitment\\ with the same source\\ wallet been already\\ posted?};
+\node[check, below right=of mfc_check3]                           (mfc_check4)  {Is \# of pending\\ redemptions on\\ the source wallet\\ equal 0?};
+\node[check, below right=of mfc_check4]                           (mfc_check5)  {Is the source\\ wallet free of pending\\ moved funds\\ sweep requests?};
+\node[check, below right=of mfc_check5]                           (mfc_check6)  {Are the\\ target wallets\\ in LIVE state?};
+\node[check, below right=of mfc_check6]                           (mfc_check7)  {Are the target\\ wallets different\\ than source\\ wallet?};
+\node[check, below right=of mfc_check7]                           (mfc_check8)  {Is \# of target\\ wallets equal to\\ min(\#\_of\_live\_wallets,\\ balance/wallet\_transfer\_max)\\ and \>0?};
+\node[check, below right=of mfc_check8]                           (mfc_check9)  {Are hashes of\\ target wallets ordered\\ ascending, without\\ duplicates?};              
+\node[revert, below left=of mfc_check9, xshift=-28cm]             (mfc_revert1) {Transaction reverted};
+\node[action, below right=of mfc_check9]                          (mfc_action1) {Information about chosen target wallets saved on-chain};
+\node[action, below=of mfc_action1]                               (mfc_action2) {Notification sent};
+\node[note1, right=of mfc_start.south east, anchor=south west]    (mfc_note1)   {Can be successfully called for wallets:\\
+                                                                                 \textbf{- MOVING\_FUNDS}};
+\node[note2, above=of mfc_note1.north, anchor=south, yshift=-8mm] (mfc_note2)   {Sender incentivized by:\\ \textbf{Fear of loosing rewards for 2 weeks; reimbursement}};
```

Comment:
Nope, they are not losing rewards but they can get slashed if they do not move funds before the timeout.

```suggestion
\node[note2, above=of mfc_note1.north, anchor=south, yshift=-8mm] (mfc_note2)   {Sender incentivized by:\\ \textbf{Ether reimbursed, avoids getting slashed}};
```

Reply:
By the `Fear of loosing rewards for 2 weeks` I meant that although the order of posting that transaction won't be enforced on-chain, operators who will notoriously skip posting the transactions on their turn (according to off-chain round-robin algorithm) can get reported by other operators in the wallet as inactive. But I can see that the way I phrased it can suggest that the loosing rewards is a direct consequence of not posting the tx. I will update it according to your suggestion.